etc/cmake/BMQPlugins.cmake

```diff
@@ -0,0 +1,86 @@
+include_guard()
+
+macro(_bmq_dedup_link_flags target)
+    cmake_parse_arguments(
+        ""
+        ""
+        "TARGET"
+        "DEPS"
+        ${ARGN}
+    )
+
+    message(DEBUG "Resolved target ${target} deps: ${${target}_DEPENDS}")
+    message(DEBUG "Culling depends from ${target}: ${_DEPS}")
+
+    bbs_import_target_dependencies(${target} ${${target}_PCDEPS})
+
+    foreach(linkopt ${${target}_DEPENDS})
+        list(FIND _DEPS "${linkopt}" idx)
+
+        if(${idx} GREATER -1)
+            if("${linkopt}" MATCHES "^-l" OR "${linkopt}" MATCHES "^[a-z]*$")
+                # Remove any link-flags like "mqb", "bmq", etc.; or any flags
+                # that are "-l<library>"; if they are already linked by
+                # 'bmqbrkr.tsk'.
+                list(REMOVE_ITEM ${target}_DEPENDS "${linkopt}")
+                if (TARGET "${linkopt}")
+                    target_include_directories(${target} PRIVATE $<TARGET_PROPERTY:${linkopt},INTERFACE_INCLUDE_DIRECTORIES>)
+                endif()
```

Comment:
Suggest adding this code to corresponding place in the enterprise repo.

Reply:
Discussed with @hallfox. `_bmq_dedup_link_flags` is a hacky solution and we **do not** want to add it to the OSS repo. Will set up meeting to discuss how we should handle it.

Reply:
We should remove the function `macro(_bmq_dedup_link_flags target)` completely.